test/analyzers.test.js

```diff
@@ -0,0 +1,210 @@
+        });
+
+        afterAll(() => {
+            videoApp.stop();
+        });
+
+        test('create an analyzer', async (done) => {
+            let postData = analyzerInfo;
+            let options =  {
+                method: 'POST',
+                uri: 'http://localhost:5000/api/v1/analyzers',
+                body: postData,
+                json: true,
+                resolveWithFullResponse: true
+            };
+            let result = await request(options);
+            expect(result.statusCode).toBe(HttpStatus.CREATED);
+            expect(result).toHaveProperty('body');
+            expect(result.body).toHaveProperty('_id');
+            analyzerId = result.body._id;
+            done();
+        });
+        // ----- test('create analyzer')
+
+        test('start the analyzer', async (done) => {
+            expect(analyzerId).not.toBeNull();
+            let options =  {
+                method: 'POST',
+                uri: 'http://localhost:5000/api/v1/analyzer/' + analyzerId + '/start',
+                json: true,
+                resolveWithFullResponse: true
+            };
+            let result = await request(options);
+
+            expect(result.statusCode).toBe(HttpStatus.NO_CONTENT);
+            done();
+        });
+        // ----- test('start the analyzer')
+
+        test('get status of the analyzer', async (done) => {
+            expect(analyzerId).not.toBeNull();
+
+            let options =  {
+                method: 'GET',
+                uri: 'http://localhost:5000/api/v1/analyzer/' + analyzerId,
+                json: true,
+                resolveWithFullResponse: true
+            };
+            let result = await request(options);
+            expect(result.statusCode).toBe(HttpStatus.CREATED);
+            expect(result).toHaveProperty('body');
+            expect(result.body).toHaveProperty('_id');
+            expect(result.body._id).toBe(analyzerId);
+            expect(result.body).toHaveProperty('name');
+            expect(result.body.name).toBe(analyzerInfo.name);
+            expect(result.body).toHaveProperty('source');
+            expect(result.body.source).toEqual(analyzerInfo.source);
+            expect(result.body).toHaveProperty('pipelines');
+            expect(result.body.pipelines).toEqual(analyzerInfo.pipelines);
+            expect(result.body).toHaveProperty('status');
+            expect(result.body.status).toBe('running');
+
+            done();
+        });
+        // ----- test('get status of the analyzer')
+
+        test('wait for notification', async (done) => {
+            let ws = new WebSocket('ws://localhost:5000/notification');
+
+                ws.on('message', function incoming(data) {
+                    try {
+                        data = data.replace(/'/g, '"');
+                        notifiedInfo = JSON.parse(data);
+                    } catch (err) {
+                        expect(err).toBeNull();
+                    }
```

Comment:
I think we should not use `try / catch` here. Under the situation without `try / catch`. If error occurs while parsing data, the test case will fail. It means that the data format is not right and we can detect the error case.